### PR TITLE
test(playwright): rename `data-cy` to `data-testid`

### PR DESCRIPTION
**Pull request by angeloashmore**
## Context

Completes DT-1839

## The Solution

- Rename `data-cy` to `data-testid` in all files.
- Remove the `testIdAttribute` Playwright setting that changed `data-testid` to `data-cy`; the default value is `data-testid`.
- Rename `dataCy` props to `testId`.

## Impact / Dependencies

N/A

## Checklist before requesting a review

- [x] I hereby declare my code ready for review.
- [x] If it is a critical feature, I have added tests.
- [ ] The CI is successful.
- [ ] If there could backward compatibility issues, it has been discussed and planned.

### CODE REVIEW

File: packages/slice-machine/test/pages/slices.test.tsx [hunk 1]
```diff
@@ -337,13 +337,17 @@ describe("slices", () => {
       },
     });
 
-    const createOneButton = document.querySelector('[data-cy="create-slice"]');
+    const createOneButton = document.querySelector(
```
Comment (xrutayisire) [inline packages/slice-machine/test/pages/slices.test.tsx]: I guess we can just getByTestId, right?

File: packages/slice-machine/test/pages/slices.test.tsx [hunk 2]
```diff
@@ -337,13 +337,17 @@ describe("slices", () => {
...
+      '[data-testid="create-slice"]',
+    );
     await act(async () => {
       // @ts-expect-error TS(2345) FIXME: Argument of type 'Element | null' is not assignabl... Remove this comment to see the full error message
       fireEvent.click(createOneButton);
     });
 
-    const nameInput = document.querySelector('[data-cy="slice-name-input"]');
+    const nameInput = document.querySelector(
```
Comment (xrutayisire) [inline packages/slice-machine/test/pages/slices.test.tsx]: I guess we can just getByTestId, right?

File: packages/slice-machine/components/ModalFormCard/index.tsx [hunk 3]
```diff
@@ -104,7 +104,7 @@ function ModalCard<Values extends FormikValues>({
           return (
             <Form
               id={formId}
-              {...(dataCy != null ? { "data-cy": dataCy } : null)}
+              {...(dataCy != null ? { "data-testid": dataCy } : null)}
```
Comment (xrutayisire) [inline packages/slice-machine/components/ModalFormCard/index.tsx]: I guess we could rename all dataCy to clean old code also, WDYT?

**Review by xrutayisire — APPROVED** [review]
Super nice, small comments for improvements. I guess if you decide not to do it, you will have a good reason so either way good to merge after.

File: packages/slice-machine/test/pages/slices.test.tsx (hunk 1 above)
Comment (angeloashmore) [reply packages/slice-machine/test/pages/slices.test.tsx]: Good catch! We already use Testing Library in this test.

File: packages/slice-machine/components/ModalFormCard/index.tsx (hunk 3 above)
Comment (angeloashmore) [reply packages/slice-machine/components/ModalFormCard/index.tsx]: Yes, good point. I didn't know about these `dataCy` props. I renamed the props to `testId`, which I think is more declarative:

```tsx
<Foo testId="foo-thing" />
```

**Pull request merged**